python-main.js

```diff
@@ -669,6 +670,9 @@ function web_editor(config) {
                     window.open('data:application/octet;charset=utf-8,' + encodeURIComponent(output), '_newtab');
                 } else {
                     var blob = new Blob([output], {type: "text/plain"});
+                    if(filename === 'main.py'){
+                        filename = scriptName + ".py";
```

Comment:
When the hex file is downloaded we replace spaces with dashes:
https://github.com/bbcmicrobit/PythonEditor/blob/7a996f1687b3989056657e81d9019e4a49aa3774/python-main.js#L724

This should probably do the same.

Might I suggest for this to create a new function called `getSafeName()` inside `web_editor()` (right below `getName()` is good), and then call that from doDownload and from this line?
And still display the `getName()` string in the user interface.

Reply:
Done